extensions/wikia/AdEngine/js/video/videoAdFactory.js

```diff
@@ -0,0 +1,60 @@
+/* global define */
+define('ext.wikia.adEngine.video.videoAdFactory', [
+	'ext.wikia.adEngine.domElementTweaker',
+	'ext.wikia.adEngine.video.googleIma',
+	'ext.wikia.adEngine.video.vastUrlBuilder',
+	'wikia.log'
+], function (DOMElementTweaker, googleIma, vastUrlBuilder, log) {
+	'use strict';
+
+	var logGroup = 'ext.wikia.adEngine.video.videoAdFactory';
+
+	function init() {
+		return googleIma.init();
+	}
+
+	function create(imageContainer, getSlotWidth, getSlotHeight, adSlot, slotParams, onVideoEndedCallback) {
+		var vastUrl = vastUrlBuilder.build(getSlotWidth() / getSlotHeight(), slotParams);
+		log(['VAST URL: ', vastUrl], log.levels.info, logGroup);
+
+		return {
+			adSlot: adSlot,
+			imageContainer: imageContainer,
+			onVideoEndedCallback: onVideoEndedCallback,
+			slotParams: slotParams,
+			videoContainer: googleIma.setupIma(vastUrl, adSlot, getSlotWidth(), getSlotHeight()),
+			play: function () {
+				var slotHeight = getSlotHeight(),
+					slotWidth = getSlotWidth(),
+					self = this;
+
+				googleIma.playVideo(slotWidth, slotHeight, function () {
+					self.toggle(true);
+					self.videoContainer = googleIma.setupIma(vastUrl, adSlot, slotWidth, slotHeight);
```

Comment:
Why do we setup again what [we've already set](https://github.com/Wikia/app/pull/11744/files#diff-11dd9e4c04b6261e40ed88433dfcf49bR25)?